### ISSUE TITLE
[Failed CI]gci-gce-ipvs CI failed 

### ISSUE THREAD

**Issue by Lion-Wei** [kubernetes/kubernetes#68560]
**What happened**:

Kube-proxy ipvs mode CI failed
http://k8s-testgrid.appspot.com/sig-network-gce#gci-gce-ipvs

But actually all test are passed in my local-up environment with the newest code.

According to ci log, those pr might be related:
- #67396
- #67097
- #67395
- #67365
- #67099

#67365 seems like the most probable, 
Anyway, looking for help.

**Comment by Lion-Wei** [comment]
Friendly ping @aleksandra-malinowska @chakri-nelluri @mengqiy @ixdy @jennybuckley , 
Please take a look, looking forward for your advise.

**Comment by Lion-Wei** [comment]
Update:
Seems like staging-k8s.gcr.io/debian-iptables-amd64:v10.1 image do have some problem, `ipset add` command will return err. Not likely to be ipset issue. 
```shell
ipset add KUBE-LOOP-BACK 192.168.0.11,udp:53,192.168.0.11
ipset v6.30: Syntax error: 'udp' is invalid as number
Syntax error: cannot parse 'udp' as a protocol
```

**Comment by Lion-Wei** [comment]
@rramkumar1 @ixdy Suggestions?

**Comment by dawidmalina** [comment]
@Lion-Wei i don't think this is strictly related to image itself.

This is result from container (image is exactly the same and host is same):
```
[malina-kubernetes-master] davidmal@malina-k8s-master0:~$ sudo docker run --rm -it --privileged docker-release.xxxx/infra/docker-kube-proxy:v1.11.3 bash
[ container ] root@2a978cbdfd6a:/$ ipset add KUBE-LOOP-BACK 192.168.0.11,udp:53,192.168.0.11
ipset v6.29: The set with the given name does not exist
```

No i will do the test within pod (kube-proxy as daemonset):
```
[malina-kubernetes-master] davidmal@malina-k8s-master0:~$ kubectl -n kube-system exec -it kube-proxy-ds-6b5dm bash
[ container ] root@malina-k8s-worker3:/$ ipset add KUBE-LOOP-BACK 192.168.0.11,udp:53,192.168.0.11
ipset v6.29: Syntax error: 'udp' is invalid as number
Syntax error: cannot parse 'udp' as a protocol
```

**Comment by ixdy** [comment]
The `ipset` version is exactly the same in both versions of the image (v10 and v10.1).

Where did you find that error message?

**Comment by ixdy** [comment]
List of package differences (given by `dpkg -l`) between v10 and v11:
```diff
@@ -5,0 +6 @@
+hi  libcap2:amd64  1:2.25-1     amd64        POSIX 1003.1e capabilities (libra
@@ -8 +9 @@
-ii  base-files     9.9+deb9u3   amd64        Debian base system miscellaneous
+ii  base-files     9.9+deb9u5   amd64        Debian base system miscellaneous
@@ -10 +11 @@
-ii  bsdutils       1:2.29.2-1   amd64        basic utilities from 4.4BSD-Lite
+ii  bsdutils       1:2.29.2-1+d amd64        basic utilities from 4.4BSD-Lite
@@ -18 +19 @@
-ii  dpkg           1.18.24      amd64        Debian package management system
+ii  dpkg           1.18.25      amd64        Debian package management system
@@ -21 +22 @@
-ii  gcc-6-base:amd 6.3.0-18     amd64        GCC, the GNU Compiler Collection
+ii  gcc-6-base:amd 6.3.0-18+deb amd64        GCC, the GNU Compiler Collection
@@ -26 +26,0 @@
-ii  inetutils-ping 2:1.9.4-2+b1 amd64        ICMP echo tool
@@ -30,0 +31 @@
+ii  iputils-ping   3:20161105-1 amd64        Tools to test the reachability of
@@ -45 +46 @@
-ii  libgcc1:amd64  1:6.3.0-18   amd64        GCC support library
+ii  libgcc1:amd64  1:6.3.0-18+d amd64        GCC support library
@@ -47,0 +49 @@
+ii  libidn11:amd64 1.33-1       amd64        GNU Libidn library, implementatio
@@ -56,0 +59 @@
+ii  libnettle6:amd 3.3-1+b2     amd64        low level cryptographic library (
@@ -66 +69 @@
-ii  libstdc++6:amd 6.3.0-18     amd64        GNU Standard C++ Library v3
+ii  libstdc++6:amd 6.3.0-18+deb amd64        GNU Standard C++ Library v3
@@ -70 +73 @@
-ii  libuuid1:amd64 2.29.2-1     amd64        Universally Unique ID library
+ii  libuuid1:amd64 2.29.2-1+deb amd64        Universally Unique ID library
@@ -76 +78,0 @@
-ii  netbase        5.4          all          Basic TCP/IP networking system
```

**Comment by Lion-Wei** [comment]
@ixdy Thanks, the error was found in ipvs ci environment kube-proxy log.

**Comment by Lion-Wei** [comment]
Maybe we got some kind of incompatible issue with ipset specific version.

**Comment by ixdy** [comment]
It seems like the removal of `netbase` is what's causing this failure, which appears to be due to depending on `iputils-ping` instead of `inetutils-ping`.

I'm not sure what caused that to switch.

**Comment by ixdy** [comment]
ok, I think I have an idea of what happened here:

`debian-base:0.3` was originally built against `debian:9.3`. At some point `debian-iptables:v10` was built against what was then `debian-base:0.3`, i.e. pulling in the packages from `debian:9.3`, which included `inetutils-ping` (and `netbase`).

Later (probably due to security issues), we rebuilt `debian-base:0.3`, which by this point was now based on `debian:9.4`. In this release, they apparently switched to installing `iputils-ping` by default, which did not include a dependency on `netbase`. We apparently did not rebuild `debian-iptables:v10`, though I'm not sure why not. As a result, it was still based on the older debian release.

By the time we built `debian-base:0.3.2`, `debian:9.5` had been released, and so `debian-iptables:v10.1` was similarly based on `debian:9.5` and was thus missing `netbase`.

tl;dr:
1. We need to add a dependency on `netbase` in `build/debian-iptables/Dockerfile` apparently, and should build/push a `debian-iptables:v11` (or maybe `v10.2`).
2. We should probably be more careful with these "simple rebases", since they can unexpectedly change what packages are installed. Some automation to do `dpkg -l` would help here. (cc @tallclair)

@Lion-Wei do you think this is a kubernetes 1.12-blocking issue?

**Comment by Lion-Wei** [comment]
@ixdy Thanks a lot. Though I'm not really understand, but glad we found the route cause. 
And I think this gonna block 1.12 release, ipvs feature already GA in 1.11, we need make sure this feature can work normally.

**Comment by Lion-Wei** [comment]
@m1093782566 @rramkumar1 for thoughts.

**Comment by m1093782566** [comment]
@rramkumar1 

Do you think this is a kubernetes 1.12-blocking issue?

**Comment by rramkumar1** [comment]
If this causing core IPVS functionality to be broken, which it looks like it is, then I think this should block the 1.12 release

**Comment by tpepper** [comment]
/milestone v1.12
Will treat as release blocking now until we hear otherwise from SIG-Network and SIG-GCP

**Comment by cblecker** [comment]
First PR has merged.

**Comment by ixdy** [comment]
Thanks. Working on promoting those images to k8s.gcr.io, though that probably won't happen till Tuesday PT.

**Comment by ixdy** [comment]
... or maybe sooner than that, because @BenTheElder is an overachiever.

**Comment by ixdy** [comment]
Second PR: #68769

**Comment by BenTheElder** [comment]
@ixdy You're the one up working on promoting images at 9pm :-) I'm just trying to set my dev environment up again and saw this going on.

**Comment by dims** [comment]
I can't keep up with the 3 of you!!! - @ixdy @BenTheElder @cblecker 

-- An old man

**Comment by Lion-Wei** [comment]
The CI working well again hours ago, thanks all of you, really appreciate it. : )

**Comment by ixdy** [comment]
Awesome.

We might want to get better monitoring on this job if IPVS is GA, since it took several weeks from the original PR merged before this issue was opened.

**Comment by BenTheElder** [comment]
Anyone willing to monitor can PR email alerts per config here: https://github.com/kubernetes/test-infra/tree/master/testgrid#email-alerts
Please consider doing so! It's ~hard~ impossible to track all of testgrid, we really need SIGs to step up on this!

**Comment by Lion-Wei** [comment]
@BenTheElder Definitely make sense, I will raise a pr to do this work.

**Comment by BenTheElder** [comment]
Thank you! 😄
